OrbitLinuxTracing/PerfEvent.h

```diff
@@ -368,11 +369,166 @@ class MapsPerfEvent : public PerfEvent {
   std::string maps_;
 };
 
-class PerfEventSampleRaw {
+class TracepointPerfEvent : public PerfEvent {
  public:
-  perf_event_sample_raw ring_buffer_record;
-  std::vector<uint8_t> data;
-  explicit PerfEventSampleRaw(uint32_t size) : data(size) {}
+  explicit TracepointPerfEvent(uint32_t size)
+      : tracepoint_data{make_unique_for_overwrite<uint8_t[]>(size)} {}
+
+  perf_event_raw_sample_fixed ring_buffer_record;
+  std::unique_ptr<uint8_t[]> tracepoint_data;
+
+  uint64_t GetTimestamp() const override {
+    return ring_buffer_record.sample_id.time;
+  }
+
+  uint64_t GetStreamId() const {
+    return ring_buffer_record.sample_id.stream_id;
+  }
+
+  uint32_t GetCpu() const { return ring_buffer_record.sample_id.cpu; }
+
+  uint16_t GetTracepointId() const { return GetTracepointCommon().common_type; }
+
+ protected:
+  const tracepoint_common& GetTracepointCommon() const {
+    return *reinterpret_cast<const tracepoint_common*>(tracepoint_data.get());
```

Comment:
We had at some point discussed the "memcpy instead of reinterpret_cast" idea. Also, std::bitcast is in C++20, so eventually we may just want to move to using that. 

Both memcpy and bit_cast are discussed a bit here:
https://en.cppreference.com/w/cpp/numeric/bit_cast

Since we have reinterpret_cast in a lot of places like this, I would say it's ok to leave as is, but we should take a close look at this at some point.

Reply:
So I'll leave it as it is now but let's discuss it offline soon as I'd also like to understand more. My understanding so far has been that since one of the types is an array of bytes then `reinterpret_cast` works correctly, but it's likely I got it wrong.

Reply:
Yes, you may be right that in this case it's actually guaranteed to work correctly.